radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/DiscoverNodesEpic.java

```diff
@@ -34,9 +35,9 @@ public DiscoverNodesEpic(Observable<RadixNode> seeds, RadixUniverseConfig config
 	public Observable<RadixNodeAction> epic(Observable<RadixNodeAction> updates, Observable<RadixNetworkState> networkState) {
 		Observable<RadixNodeAction> getSeedUniverses = updates
 			.ofType(DiscoverMoreNodesAction.class)
-			.firstOrError()
-			.flatMapObservable(i -> seeds)
-			.map(GetUniverseRequestAction::of);
+			.flatMap(i -> seeds)
+			.<RadixNodeAction>map(GetUniverseRequestAction::of)
+			.onErrorReturn(DiscoverMoreNodesErrorAction::new);
```

Comment:
Is this `onErrorReturn(...)` for not breaking the observable stream (as a hard `onError` would) and thereby saving me the effort of having to "reconnect" after each error?

Beautiful, if so!

Reply:
Yup, exactly. It's RX version of try/catch.